# check the quality of RNAseq reads using narval

# raw reads
# 2022.03.29
# in narval


```
module load StdEnv/2020
module load fastqc/0.11.9

cd $SCRATCH/2022_05_Tfucif_RNAseq/00_RawData
fastqc -o ../01_QC/pre_trimming_QC -t 8 *_R1.fastq *_R2.fastq

module load StdEnv/2020
module load fastqc/0.11.9

cd $SCRATCH/2022_05_Tfucif_RNAseq/00_RawData
fastqc -o ../01_QC/pre_trimming_QC -t 8 *_R1.fastq *_R2.fastq
```

move files to debary

install multiQC into conda environment multiqc
MultiQC v1.12

```
cd /data/ccallen/2022_05_Tfuci_RNAseq/01_QC/pre_trimming_QC

multiqc .
```

information on which sequences need to be trimmed from reads:

https://international.neb.com/faqs/2021/01/15/what-sequences-need-to-be-trimmed-for-nebnext-libraries-that-are-sequenced-on-an-illumina-instrument

This will perform the following:

Remove adapters (ILLUMINACLIP:TruSeq3-PE.fa:2:30:10)
Remove leading low quality or N bases (below quality 3) (LEADING:3)
Remove trailing low quality or N bases (below quality 3) (TRAILING:3)
	Scan the read with a 4-base wide sliding window, cutting when the average quality per base drops below 15 (SLIDINGWINDOW:4:15)
	Drop reads below the 36 bases long (MINLEN:36)

```
java -jar /data/ccallen/miniconda/envs/trimmomatic-env/share/trimmomatic-0.39-1/trimmomatic.jar PE \
        -phred33 \
        "$i"_1.fastq.gz  "$i"_2.fastq.gz "$i"_1_trimmed.fastq.gz "$i"_1_unpaired.fastq.gz "$i"_2_trimmed.fastq.gz "$i"_2_unpaired.fastq.gz \
        ILLUMINACLIP:/data/ccallen/miniconda/envs/trimmomatic-env/share/trimmomatic-0.39-1/adapters/TruSeq3-PE-2.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
```
```
java -jar /data/ccallen/miniconda/envs/trimmomatic-env/share/trimmomatic-0.39-1/trimmomatic.jar PE \
        -phred33 \
        00_RawData/NS.1829.004.NEBNext_dual_i7_A3---NEBNext_dual_i5_A3.CA202112_R1.fastq.gz \
        00_RawData/NS.1829.004.NEBNext_dual_i7_A3---NEBNext_dual_i5_A3.CA202112_R2.fastq.gz \
        03_CLEAN_READS/ca202112_1_trimmed.fastq.gz \
        03_CLEAN_READS/ca202112_1_unpaired.fastq.gz \
        03_CLEAN_READS/ca202112_2_trimmed.fastq.gz \
        03_CLEAN_READS/ca202112_2_unpaired.fastq.gz \
        ILLUMINACLIP:/data/ccallen/miniconda/envs/trimmomatic-env/share/trimmomatic-0.39-1/adapters/TruSeq3-PE-2.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
```

https://web.azenta.com/a-quick-start-guide-to-rna-seq-data-analysis

https://international.neb.com/faqs/2021/01/15/what-sequences-need-to-be-trimmed-for-nebnext-libraries-that-are-sequenced-on-an-illumina-instrument

https://github.com/usadellab/Trimmomatic
```
java -jar trimmomatic-0.39.jar -phred33 PE input_forward.fastq.gz input_reverse.fastq.gz output_forward_paired.fastq.gz output_forward_unpaired.fastq.gz output_reverse_paired.fastq.gz output_reverse_unpaired.fastq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36

```

The displayed qualities are the phred quality scores. Since your data was sequenced on the NovaSeq 6000, they use the RTA3 binned score scheme. (The only observable Phred quality scores are  F |  :  |  ,  | #  which respectively correspond to 37 | 25 | 11 | 2 )

TNATCGTTGATATCCCGATCCATATTGCTATCCTGTTCTTTGCGTTCGCGCAAGCGCTTCCAGGCGGCGGTTTTCTCAAACGCAATGCCAGAAACCTTTAG