# check the quality of RNAseq reads using narval

# raw reads
# 2022.03.29
# in narval


```
module load StdEnv/2020
module load fastqc/0.11.9

cd $SCRATCH/2022_05_Tfucif_RNAseq/00_RawData
fastqc -o ../01_QC/pre_trimming_QC -t 8 *_R1.fastq *_R2.fastq

module load StdEnv/2020
module load fastqc/0.11.9

cd $SCRATCH/2022_05_Tfucif_RNAseq/00_RawData
fastqc -o ../01_QC/pre_trimming_QC -t 8 *_R1.fastq *_R2.fastq
```

move files to debary

install multiQC into conda environment multiqc
MultiQC v1.12

```
cd /data/ccallen/2022_05_Tfuci_RNAseq/01_QC/pre_trimming_QC

multiqc .
```

information on which sequences need to be trimmed from reads:

https://international.neb.com/faqs/2021/01/15/what-sequences-need-to-be-trimmed-for-nebnext-libraries-that-are-sequenced-on-an-illumina-instrument

This will perform the following:

Remove adapters (ILLUMINACLIP:TruSeq3-PE.fa:2:30:10)
Remove leading low quality or N bases (below quality 3) (LEADING:3)
Remove trailing low quality or N bases (below quality 3) (TRAILING:3)
	Scan the read with a 4-base wide sliding window, cutting when the average quality per base drops below 15 (SLIDINGWINDOW:4:15)
	Drop reads below the 36 bases long (MINLEN:36)

https://web.azenta.com/a-quick-start-guide-to-rna-seq-data-analysis

https://international.neb.com/faqs/2021/01/15/what-sequences-need-to-be-trimmed-for-nebnext-libraries-that-are-sequenced-on-an-illumina-instrument

https://github.com/usadellab/Trimmomatic
```

#!/bin/bash

#SBATCH --account=def-tspribi
#SBATCH --time=2-0:0
#SBATCH --cpus-per-task=8
#SBATCH --mem=8G
#SBATCH --job-name=trim
#SBATCH --output=trim.script.logs.out
#SBATCH --mail-user=w6p9c9j6t9c6a2i6@spribillelabworkspace.slack.com
#SBATCH --mail-type=BEGIN
#SBATCH --mail-type=END

module load StdEnv/2020
module load trimmomatic/0.39

cd /scratch/ccallen/2022_05_Tfucif_RNAseq/02_CLEAN_READS
for sample in $(cat ../samples.txt)
do
java -jar $EBROOTTRIMMOMATIC/trimmomatic-0.39.jar PE -threads 8 -phred33 ../00_RawData/"$sample"_R1.fastq ../00_RawData/"$sample"_R2.fastq "$sample"_R1_paired.fastq "$sample"_R1_unpaired.fastq "$sample"_R2_paired.fastq "$sample"_R2_unpaired.fastq.gz ILLUMINACLIP:$EBROOTTRIMMOMATIC/adapters/TruSeq3-PE-2.fa:2:30:10:2:True LEADING:3 TRAILING:3 MINLEN:36
done

```

The displayed qualities are the phred quality scores. Since your data was sequenced on the NovaSeq 6000, they use the RTA3 binned score scheme. (The only observable Phred quality scores are  F |  :  |  ,  | #  which respectively correspond to 37 | 25 | 11 | 2 )


# less sensitive trimming

This will perform the following:

Remove adapters (ILLUMINACLIP:TruSeq3-PE.fa:2:30:10)
Remove leading low quality or N bases (below quality 3) (LEADING:3)
Remove trailing low quality or N bases (below quality 3) (TRAILING:3)
	Scan the read with a 4-base wide sliding window, cutting when the average quality per base drops below 15 (SLIDINGWINDOW:4:15)
	Drop reads below the 36 bases long (MINLEN:36)

```[ccallen@narval2 relaxed]$ cat ../../scripts/run_trimmomatic_relaxed.sh
#!/bin/bash

#SBATCH --account=def-tspribi
#SBATCH --time=2-0:0
#SBATCH --cpus-per-task=8
#SBATCH --mem=8G
#SBATCH --job-name=trim
#SBATCH --output=trim_relaxed.script.logs.out
#SBATCH --mail-user=w6p9c9j6t9c6a2i6@spribillelabworkspace.slack.com
#SBATCH --mail-type=BEGIN
#SBATCH --mail-type=END

module load StdEnv/2020
module load trimmomatic/0.39

cd /scratch/ccallen/2022_05_Tfucif_RNAseq/02_CLEAN_READS/relaxed
for sample in $(cat ../../samples.txt)
do
java -jar $EBROOTTRIMMOMATIC/trimmomatic-0.39.jar PE -threads 8 -phred33 ../../00_RawData/"$sample"_R1.fastq ../../00_RawData/"$sample"_R2.fastq "$sample"_R1_paired.fastq "$sample"_R1_unpaired.fastq "$sample"_R2_paired.fastq "$sample"_R2_unpaired.fastq ILLUMINACLIP:$EBROOTTRIMMOMATIC/adapters/TruSeq3-PE-2.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
done
```
# check QC of trimmed reads, relaxed and strict
```
#!/bin/bash

#SBATCH --account=def-tspribi
#SBATCH --time=2-0:0
#SBATCH --cpus-per-task=8
#SBATCH --mem=8G
#SBATCH --job-name=postQC
#SBATCH --output=post_filter.script.logs.out
#SBATCH --mail-user=w6p9c9j6t9c6a2i6@spribillelabworkspace.slack.com
#SBATCH --mail-type=BEGIN
#SBATCH --mail-type=END

module load StdEnv/2020
module load fastqc/0.11.9

cd $SCRATCH/2022_05_Tfucif_RNAseq/02_CLEAN_READS/strict
fastqc -o ../../01_QC/post_trimming_QC/strict -t 8 *_paired.fastq

cd $SCRATCH/2022_05_Tfucif_RNAseq/02_CLEAN_READS/relaxed
fastqc -o ../../01_QC/post_trimming_QC/relaxed -t 8 *_paired.fastq
```

```
#!/bin/bash

#SBATCH --account=def-tspribi
#SBATCH --time=2-0:0
#SBATCH --cpus-per-task=8
#SBATCH --mem=8G
#SBATCH --job-name=trim
#SBATCH --output=trim_relaxed_keep.script.logs.out
#SBATCH --mail-user=w6p9c9j6t9c6a2i6@spribillelabworkspace.slack.com
#SBATCH --mail-type=BEGIN
#SBATCH --mail-type=END

module load StdEnv/2020
module load trimmomatic/0.39

cd /scratch/ccallen/2022_05_Tfucif_RNAseq/02_CLEAN_READS/relaxed_keep
for sample in $(cat ../../samples.txt)
do
java -jar $EBROOTTRIMMOMATIC/trimmomatic-0.39.jar PE -threads 8 -phred33 ../../00_RawData/"$sample"_R1.fastq ../../00_RawData/"$sample"_R2.fastq "$sample"_R1_paired.fastq "$sample"_R1_unpaired.fastq "$sample"_R2_paired.fastq "$sample"_R2_unpaired.fastq ILLUMINACLIP:$EBROOTTRIMMOMATIC/adapters/TruSeq3-PE-2.fa:2:30:10:2:True LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
done
```
